Client/Frontend/Settings/BraveRewardsSettingsViewController.swift

```diff
@@ -47,25 +52,53 @@ class BraveRewardsSettingsViewController: TableViewController {
             )
         ]
         
-        if let legacyWallet = legacyWallet, !legacyWallet.isLedgerTransferExpired {
-            legacyWallet.transferrableAmount({ [weak self] total in
-                guard let self = self, total > 0 else { return }
-                self.dataSource.sections.insert(.init(
-                    header: .title(Strings.Rewards.walletTransferTitle),
-                    rows: [
-                        Row(text: Strings.Rewards.legacyWalletTransfer, selection: { [unowned self] in
-                            guard let legacyWallet = self.legacyWallet else { return }
-                            let controller = WalletTransferViewController(legacyWallet: legacyWallet)
-                            controller.learnMoreHandler = { [weak self] in
-                                self?.walletTransferLearnMoreTapped?()
-                            }
-                            let container = UINavigationController(rootViewController: controller)
-                            container.modalPresentationStyle = .formSheet
-                            self.present(container, animated: true)
-                        }, image: UIImage(imageLiteralResourceName: "rewards-qr-code").template)
-                    ]
-                ), at: 1)
-            })
+        if let legacyWallet = legacyWallet {
+            if Preferences.Rewards.transferDrainID.value == nil {
+                if !legacyWallet.isLedgerTransferExpired {
+                    legacyWallet.transferrableAmount({ [weak self] total in
+                        guard let self = self, total > 0 else { return }
+                        self.dataSource.sections.insert(.init(
+                            header: .title(Strings.Rewards.walletTransferTitle),
+                            rows: [
+                                Row(text: Strings.Rewards.legacyWalletTransfer,
+                                    detailText: Preferences.Rewards.lastTransferStatus.value.map(DrainStatus.init)??.displayString,
```

Comment:
lol nice magic here, is it equal to `DrainStatus(rawValue: lastTransferStatus.value)`?

Reply:
Yeah `map` on an optional type returns a type based on the value if it's not nil, but unfortunately a `RawRepresentable` init is also optional, thus the double `?`